server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -443,6 +449,54 @@ TransportResponse handleExtensionRequest(ExtensionRequest extensionRequest) thro
         }
     }
 
+    /**
+     * A separate transport action handler used to issue service accounts to extensions during initialization
+     * @param extension The extension to be issued a service account
+     */
+    public void issueServiceAccount(Extension extension) {
+        DiscoveryExtensionNode discoveryExtensionNode = extensionIdMap.get(extension.getUniqueId());
+        AuthToken serviceAccountToken = identityService.getTokenManager().issueServiceAccountToken(extension.getUniqueId());
+        String authTokenAsString = serviceAccountToken.asAuthHeaderValue();
+        final CompletableFuture<IssueServiceAccountResponse> inProgressFuture = new CompletableFuture<>();
```

Comment:
This object it isn't passed into any methods to be waited on, so it seems like we can remove it?

Reply:
Note; I think this is still the case in the initialization flow and it was copy/pasted here.

Reply:
Yeah will do. I just copied from the other flow, but I will remove it.